src/main/java/org/generationcp/middleware/dao/GermplasmStudySourceDAO.java

```diff
@@ -0,0 +1,195 @@
+package org.generationcp.middleware.dao;
+
+import liquibase.util.StringUtils;
+import org.generationcp.middleware.pojos.GermplasmStudySource;
+import org.generationcp.middleware.pojos.SortedPageRequest;
+import org.generationcp.middleware.service.api.study.germplasm.source.StudyGermplasmSourceDto;
+import org.generationcp.middleware.service.api.study.germplasm.source.StudyGermplasmSourceRequest;
+import org.generationcp.middleware.util.SqlQueryParamBuilder;
+import org.hibernate.Criteria;
+import org.hibernate.SQLQuery;
+import org.hibernate.criterion.Projections;
+import org.hibernate.criterion.Restrictions;
+import org.hibernate.transform.Transformers;
+import org.hibernate.type.IntegerType;
+
+import java.math.BigInteger;
+import java.util.List;
+
+public class GermplasmStudySourceDAO extends GenericDAO<GermplasmStudySource, Integer> {
+
+	protected static final String GERMPLASM_STUDY_SOURCE_SEARCH_QUERY = "SELECT \n"
+		+ "gss.source_id as `sourceId`,\n"
+		+ "gss.gid as `gid`,\n"
+		+ "g.mgid as `groupId`,\n"
+		+ "n.nval as `designation`,\n"
+		+ "m.mcode as `breedingMethodAbbreviation`,\n"
+		+ "m.mname as `breedingMethodName`,\n"
+		+ "m.mtype as `breedingMethodType`,\n"
+		+ "geo.description as `trialInstance`,\n"
+		+ "loc.lname as `location`,\n"
+		+ "rep_no.value as `replicationNumber`,\n"
+		+ "plot_no.value as `plotNumber`,\n"
+		+ "g.gdate as `germplasmDate`,\n"
+		+ "count(lot.lotid) as `lots` "
+		+ "FROM germplasm_study_source gss \n"
+		+ "INNER JOIN germplsm g ON g.gid = gss.gid\n"
+		+ "INNER JOIN project p ON p.project_id = gss.project_id\n"
+		+ "LEFT JOIN nd_experiment e ON e.nd_experiment_id = gss.nd_experiment_id\n"
+		+ "LEFT JOIN nd_experimentprop rep_no ON rep_no.nd_experiment_id = e.nd_experiment_id AND rep_no.type_id = 8210\n"
+		+ "LEFT JOIN nd_experimentprop plot_no ON plot_no.nd_experiment_id = e.nd_experiment_id AND plot_no.type_id = 8200\n"
+		+ "LEFT JOIN nd_geolocation geo ON e.nd_geolocation_id = geo.nd_geolocation_id\n"
+		+ "LEFT JOIN nd_geolocationprop locationProp ON geo.nd_geolocation_id = locationProp.nd_geolocation_id AND locationProp.type_id = 8190\n"
+		+ "LEFT JOIN location loc ON locationProp.value = loc.locid\n"
+		+ "LEFT JOIN methods m ON m.mid = g.methn\n"
+		+ "LEFT JOIN names n ON g.gid = n.gid AND n.ntype in (2, 5)\n"
+		+ "LEFT JOIN ims_stock_transaction stockTransaction ON gss.source_id  = stockTransaction.source_id\n"
+		+ "LEFT JOIN ims_transaction transaction ON stockTransaction.trnid = transaction.trnid\n"
+		+ "LEFT JOIN ims_transaction lot ON  transaction.lotid = lot.lotid\n"
+		+ "WHERE gss.project_id = :studyId ";
+
+	public List<StudyGermplasmSourceDto> getGermplasmStudySourceList(final StudyGermplasmSourceRequest studyGermplasmSourceRequest) {
```

Comment:
Renaem StudyGermplasmSourceXXX classes GermplasmStudySourceXXX for consistency

Reply:
Done. I updated all the class names, parameters in java and js files.